termwiz/src/tmux_cc/mod.rs

```diff
@@ -716,4 +829,58 @@ here
             events
         );
     }
+
+    #[test]
+    fn test_parse_layout() {
+        let layout_case1 = "158x40,0,0,72".to_string();
+        let layout_case2 = "158x40,0,0[158x20,0,0,69,158x19,0,21{79x19,0,21,70,78x19,80,21[78x9,80,21,71,78x9,80,31,73]}]".to_string();
+        let layout_case3 = "158x40,0,0{79x40,0,0[79x20,0,0,74,79x19,0,21{39x19,0,21,76,39x19,40,21,77}],78x40,80,0,75}".to_string();
+
+        let mut layout = parse_layout(layout_case1.get(0..).unwrap()).unwrap();
+        let l = layout.pop().unwrap();
+        assert!(if let WindowLayout::SinglePane(p) = l {
+            assert_eq!(p.pane_width, 158);
+            assert_eq!(p.pane_height, 40);
+            assert_eq!(p.pane_left, 0);
+            assert_eq!(p.pane_top, 0);
+            assert_eq!(p.pane_id, 72);
+            true
+        } else {
+            false
+        });
+
+        layout = parse_layout(layout_case2.get(0..).unwrap()).unwrap();
+        assert!(if let WindowLayout::SplitVertical(_x) = &layout[0] {
+            true
+        } else {
+            false
+        });
```

Comment:
you can rewrite these like this:

```suggestion
        assert!(matches!(&layout[0], WindowLayout::SplitVertical(_x)));
```